### ISSUE TITLE
Rewrite NCONF

### ISSUE THREAD

**Issue by richsalz** [openssl/openssl#11662]
The NCONF implemention needs to be rewritten.  Right now it's based on CONF and ```_CONF```

The whole thing should be cleaned up, there are probably corner-cases of escaping and such to be fixed. There is definitely multi-directory scanning limitation.  The documentation was rewritten for 3.0, it points out these bugs (and other issues).

**Comment by levitte** [comment]
Agreed.  It's hard to rewrite an existing API, though, so I'd rather think that a new API would be more suitable.  Following our prefix scheme for new APIs, I would suggest `OSSL_CONF`...

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of 3.4  dev, barring further input

**Comment by t8m** [comment]
Closing as the issue was labeled as inactive for extended period of time and there was no additional input.
Please add a comment if you have any further information to support reopening the issue.
